internal/pkg/statemanager/state.go

```diff
@@ -0,0 +1,63 @@
+package statemanager
+
+import (
+	"bytes"
+	"log"
+
+	"gopkg.in/yaml.v3"
+)
+
+type CompStatus string
+type CompAction string
+
+const (
+	StatusUninstalled CompStatus = "uninstalled"
+	StatusInstalled   CompStatus = "installed"
+	StatusRunning     CompStatus = "running"
+	StatusFailed      CompStatus = "Failed"
+)
+
+const (
+	ActionInstall   CompAction = "install"
+	ActionReinstall CompAction = "reinstall"
+	ActionUninstall CompAction = "uninstall"
+)
+
+type States map[string]*State
+
+// State is the single component's state.
+type State struct {
+	Name          string     `yaml:"name"`
+	Version       string     `yaml:"version"`
+	Dependencies  []string   `yaml:"dependencies"`
+	Status        CompStatus `yaml:"status"`
+	LastOperation *Operation `yaml:"lastOperation"`
+}
+
```

Comment:
For better consistency (at least within the same package), we should either add comments for all exported variables/functions, or none at all (by using self-explanatory names).

Reply:
I tend to write more comments, but considering the workload, I hope to implement features first, and then gradually improve unit tests and comments